### PR TITLE
dumpling: solve inconsistent capitalization of column name in function GetSpecifiedColumnValuesAndClose when being compared

### LINKED ISSUE #38341: Capitalization of column name in function GetSpecifiedColumnValuesAndClose should be consistent when being compared
https://github.com/pingcap/tidb/blob/fd468fffac0a545dc4a2a13eeb93ecf8d10735c7/dumpling/export/sql.go#L693-L736

In the function GetSpecifiedColumnValuesAndClose (line 710):
```
if strings.ToUpper(col) == name
```
The column name `col` is converted to the upper case and then compared with `name`, where `col` and `name`are inconsistent in the capitalization. 

Suggested solutions:
`col` should not be converted to the upper case, or `name` should also be converted to the upper case before the comparison.

### PR DESCRIPTION
**Pull request by lyzx2001**
### What problem does this PR solve?

Issue Number: close #38341 

Problem Summary:
https://github.com/pingcap/tidb/blob/fd468fffac0a545dc4a2a13eeb93ecf8d10735c7/dumpling/export/sql.go#L693-L736

In the function GetSpecifiedColumnValuesAndClose (line 710):
```
if strings.ToUpper(col) == name
```
The column name `col` is converted to the upper case and then compared with `name`, where `col` and `name`are inconsistent in the capitalization. 

### What is changed and how it works?
`name` is also converted to upper case before the comparison with `col`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lichunzhu — APPROVED** [review]
LGTM

File: dumpling/export/sql.go [hunk 1]
```diff
@@ -707,7 +707,7 @@ func GetSpecifiedColumnValuesAndClose(rows *sql.Rows, columnName ...string) ([][
 	for i, col := range columns {
 		addr[i] = &oneRow[i]
 		for j, name := range columnName {
-			if strings.ToUpper(col) == name {
+			if strings.EqualFold(strings.ToUpper(col), strings.ToUpper(name)) {
```
Comment (lance6716) [inline dumpling/export/sql.go:710]:
```suggestion
			if strings.EqualFold(col, name) {
```

EqualFold is more than case-insensitive.

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

**Pull request merged**